File: src/main/java/ROT13.java
public class ROT13 {
    public String encrypt(String sIn) {
        //BEGIN_CHALLENGE
        char[] aIn = sIn.toCharArray();
        for (int i = 0; i < aIn.length; i++) {

            // Filter 'out of range' characters
            if (aIn[i] >= 'A' && aIn[i] <= 'Z' || aIn[i] >= 'a' && aIn[i] <= 'z') {
                // Apply fix for A-Z or a-z
                if ((aIn[i] <= 'Z' ? aIn[i] - 'A' : aIn[i] - 'a') < 13)
                    aIn[i] += 13;
                else
                    aIn[i] -= 13;
            }
        }
        sIn = new String(aIn);
        //END_CHALLENGE
        return sIn;
    }
}

File: src/ROT13.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rot13Challenge.src
{
    public class ROT13
    {
        //BEGIN_CHALLENGE
        private char[] LookupTable = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".ToCharArray();
        private char[] TransformTable = "NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm".ToCharArray();
        //END_CHALLENGE

        public string Encrypt(string sIn)
        {
            // Write your code below here
            //BEGIN_CHALLENGE
            sIn = new string(sIn
                .Select(x => {
                    if (LookupTable.Contains(x))
                        return TransformTable[Array.IndexOf(LookupTable, x)];
                    return x;
                })
                .ToArray());
            //END_CHALLENGE
            return sIn;
        }
    }
}
